### DISCUSSION TITLE
when I use fzf extra, Keymap are not available for use under the project extra

### DISCUSSION THREAD

**Discussion by gengdz** [LazyVim/LazyVim#3665] [Q&A]
I have mapped `CTRL-T` to open the selected item in a new tab, 

which works when using `leader ff` or `leader sg`. 

However, when I select a project with `leader fp` and then attempt to use `CTRL-T`, it doesn't work as expected. 

Upon examining the source code, I found that the issue arises because the logic for the default action within the 'project extra',No other shortcut keys defined, resulting in failure to take effect

 What steps should I take to rectify this?

I am sorry that I'm currently editing via my mobile device, hence I'm unable to provide screenshots or additional information. I hope the explanation is clear enough.

**Comment by folke** [comment]
you can provide a PR for the extra?

**Reply by gengdz** [reply]
This is quite a challenge for me, but I can give it a try. 

I've never written in Lua before; I am not familiar with Neovim-related APIs.

I mainly use front-end related languages.

**Reply by folke** [reply]
Added: https://github.com/LazyVim/LazyVim/commit/f28937b0c53195a90fb450d0f07bacd3321a0cfd

**Reply by gengdz** [reply]
good~

I have some ideas：
If this is how to solve the problem, then I indeed achieved it yesterday(I can imitate some code, but I can't creatively write a large amount of code.)

My idea of a solution is:
In my opinion, project extra should inherit all of the user's custom key bindings, rather than explicitly defining a certain key, such as `<C-t>` to open in a new tab.

Because users might want to redefine other key bindings here. If there is such a need, then modifying the code of LazyVim would be necessary. I think this might not be a good way to do it.

**Comment by dpetka2001** [comment]
How do you override the keymap. In `fzf-lua.files` `<ctrl-t>` is by default mapped to open Trouble. You would have to override that in your personal config. The following should work
```lua
return {
  {
    "ibhagwan/fzf-lua",
    opts = function(_, opts)
      local config = require("fzf-lua.config")
      local actions = require("fzf-lua.actions")

      config.defaults.actions.files["ctrl-t"] = actions.file_tabedit
    end,
  }
}
```

**Reply by gengdz** [reply]
Your code is exactly what I meant by saying「I have mapped `C-t` to open the selected item in a new tab, which works when using `<leader> ff` or `<leader> sg`」

However, it does not work like this in `<leader> fp`

**Reply by dpetka2001** [reply]
Strange, I've just tried it and it actually opens the file in a new tab for me.

**Reply by gengdz** [reply]
I guess your operation should be like this:

Suppose you have 3 projects: project-a, project-b, project-c

1. You use `<leader>fp`, and then 3 projects appear.
2. You search or navigate to one of them, then use `<Enter>` to enter into a specific project.
3. Then you use `C-t` to open this project in a new tab.

With these steps, it indeed works.

My expected steps are:
1. use `<leader>fp`, and then 3 projects appear
2. search or navigate to one of them， use `C-t` to open this project in a new tab

**Reply by folke** [reply]
feel free to provide a pr. 
you need to add the proper action in the project.lua extra. see that code

**Reply by dpetka2001** [reply]
Yes, I used `<CR>` to enter into the project, then fzf-lua shows the files available in that project and when I hit `<ctrl-t>` it opens the file in a new tab. I don't use the `util.project` Extra myself, just did this for testing purposes.
